cognite_toolkit/_cdf_tk/commands/_upload.py

```diff
@@ -86,8 +92,50 @@ def upload(
 
         self._deploy_resource_folder(input_dir / DATA_RESOURCE_DIR, deploy_resources, client, console, dry_run, verbose)
 
+        data_files_by_selector = self._prepare(data_files_by_selector, client)
```

Comment:
Maybe this can have a more specific name, as it is a special case.

```suggestion
        data_files_by_selector = self._topological_sort_if_instance_selector(data_files_by_selector, client)
```